src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrWoman = (!withChildren)
+    ? people.filter((human) => human.sex === 'f')
+    : people.filter((mother) => mother.sex === 'f'
+      && (people.find((person) => mother.name === person.mother)));
+
+  const averageAgeWonam = arrWoman.map((person) => person.died - person.born)
+    .reduce((a, b) => a + b) / arrWoman.length;
+
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.

As an option(up to you) you could create 2 separate variables: `women` and `womenWithChildren` and pass them as arguments to call your new `getAverage` function. This way ternary operator would be in return statement and code would be more clear